packages/kbn-router-to-openapispec/src/extract_authz_description.test.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { schema } from '@kbn/config-schema';
+import { extractAuthzDescription } from './extract_authz_description';
+import { InternalRouterRoute } from './type';
+
+describe('extractAuthzDescription', () => {
+  it('should return empty if route does not require privileges', () => {
+    const route: InternalRouterRoute = {
+      path: '/foo',
+      options: { access: 'internal' },
+      handler: jest.fn(),
+      validationSchemas: { request: { body: schema.object({}) } },
+      method: 'get',
+      isVersioned: false,
+    };
+    const description = extractAuthzDescription(route.security);
+    expect(description).toBe('');
+  });
+
+  it('should return route authz description for simple privileges', () => {
+    const route: InternalRouterRoute = {
+      path: '/foo',
+      options: { access: 'public' },
+      handler: jest.fn(),
+      validationSchemas: { request: { body: schema.object({}) } },
+      method: 'get',
+      isVersioned: false,
+      security: {
+        authz: {
+          requiredPrivileges: ['manage_spaces'],
+        },
+      },
+    };
+    const description = extractAuthzDescription(route.security);
+    expect(description).toBe('[Authz] Route required privileges: ALL of [manage_spaces].');
+  });
+
+  it('should return route authz description for privilege groups', () => {
+    {
+      const route: InternalRouterRoute = {
+        path: '/foo',
+        options: { access: 'public' },
+        handler: jest.fn(),
+        validationSchemas: { request: { body: schema.object({}) } },
+        method: 'get',
+        isVersioned: false,
+        security: {
+          authz: {
+            requiredPrivileges: [{ allRequired: ['console'], anyRequired: ['manage_spaces'] }],
```

Comment:
nit: maybe we should have a test case with only `allRequired`, one with only `anyRequired` and one with both. WDYT?